### DISCUSSION TITLE
Any plan to support mirai-api-http 2.X?

### DISCUSSION THREAD

**Discussion by mice2100** [YunYouJun/mirai-ts#36] [General]
Seems 1.x has been frozen. 2.x is now active

**Comment by YunYouJun** [comment]
Of course. But I am working overtime at the company recently, so the work will start later.

In addition to the factors, there is also a part of waiting for the stability of 2.0.

**Comment by YunYouJun** [comment]
[mirai-ts@2.x](https://github.com/YunYouJun/mirai-ts/tree/2.x) is working! You can install `mirai-ts@next` to try it.

**Comment by mice2100** [comment]
请教：
我在使用mirai-ts@next的时候，里面有部分API用法不太理解。比如下面代码：
  const glist = await miraiInst.api.groupList()
  console.log(glist)
  if(glist) {
    for(const g of glist) {

实际console打印出来的是HTTP-API返回的原始格式，而不是里面的data。
请问正确的用法应该怎么写？

**Reply by YunYouJun** [reply]
Try this.

```ts
const { data } = await mirai.api.groupList()
```

**Reply by mice2100** [reply]
Won't work. ESLint shows:
Property 'data' does not exist on type 'FriendList'.ts(2339)

**Reply by mice2100** [reply]
I checked the source file and seems there is a mismatch:
  async groupList(): Promise<Api.Response.GroupList> {
    const { data } = await this.axios.get("/groupList", {
      params: {
        sessionKey: this.sessionKey,
      },
    });
    return data; //should be data.data
  }

**Comment by YunYouJun** [comment]
~~笑死，都是中国人，我就不说英文了~~

我发现 mirai-api-http 的 API 并没有遵循统一的规范，有的是有 `code` 有的则没有，这我得处理一下。

**Comment by YunYouJun** [comment]
好友列表类型问题已修复，released in `2.0.0-alpha.6`。

可以尝试如下使用方式。

```ts
const { data } = await mirai.api.groupList()
```

mirai-api-http 文档很多地方规范没有统一，且会变动，我没法一一对照更新。

再有新的问题欢迎反馈和 PR。
